cmd/exo/cmd/k8s_create.go

```diff
@@ -0,0 +1,381 @@
+
+	rule, err := getDefaultRule(SSH.String(), false)
+	if err != nil {
+		return err
+	}
+
+	rule.SecurityGroupID = securityGroupID
+
+	if err := addRule(rule, false); err != nil {
+		return err
+	}
+
+	rule, err = getDefaultRule(ETCDCLIENT.String(), false)
+	if err != nil {
+		return err
+	}
+
+	rule.SecurityGroupID = securityGroupID
+
+	if err := addRule(rule, false); err != nil {
+		return err
+	}
+
+	rule, err = getDefaultRule(ETCDSERVER.String(), false)
+	if err != nil {
+		return err
+	}
+
+	rule.SecurityGroupID = securityGroupID
+
+	if err := addRule(rule, false); err != nil {
+		return err
+	}
+
+	rule, err = getDefaultRule(RKEHTTPS.String(), false)
+	if err != nil {
+		return err
+	}
+
+	rule.SecurityGroupID = securityGroupID
+
+	return addRule(rule, false)
+}
+
+func storeConfig(clusterName, clusterFile string, nodes []string) error {
+	filePath := path.Join(configFolder, "k8s", "clusters", clusterName)
+
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		if err := os.MkdirAll(filePath, os.ModePerm); err != nil {
+			return err
+		}
+	}
+
+	filePath = path.Join(filePath, clusterFileName)
+
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		if err := ioutil.WriteFile(filePath, []byte(clusterFile), 0600); err != nil {
+			log.Fatalf("cluster.yml could not be written. %s", err)
+		}
+	}
+
+	nodesFile := strings.Join(nodes, "\n")
+
+	filePath = path.Join(configFolder, "k8s", "clusters", clusterName, "nodes")
+
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		if err := ioutil.WriteFile(filePath, []byte(nodesFile), 0600); err != nil {
+			log.Fatalf("nodes could not be written. %s", err)
+		}
+	}
+	return nil
+}
+
+func checkingCloudInitJob(vms []string) error {
+
+	defer println("")
+
+	print("Installing Docker on node(s)")
+	for i := 0; i < 120; i++ {
+
+		var errCMD error
+		for _, vm := range vms {
+
+			sshinfo, err := getSSHInfos(vm)
+			if err != nil {
+				return err
+			}
+
+			args := []string{
+				fmt.Sprintf("%s@%s", sshinfo.userName, sshinfo.ip.String()),
+				"-i",
+				sshinfo.sshKeys,
+				"-o",
+				"StrictHostKeyChecking=no",
+				"docker",
+				"ps",
+			}
+
+			cmd := exec.Command("ssh", args...)
+
+			errCMD = cmd.Run()
+			if errCMD != nil {
+				break
+			}
+		}
+		if errCMD == nil {
+			return nil
+		}
+		ttime.Sleep(ttime.Second)
+		print(".")
+		errCMD = nil
+	}
+
+	return fmt.Errorf("waiting to installing Docker Timeout")
+}
+
+func destroyAllNodes(nodes []string) error {
+	for _, node := range nodes {
+		if err := deleteVM(node); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func deployNodes(nodeNumber int, nodeCapacity, sg string) ([]string, error) {
+
+	nodes := make([]string, nodeNumber)
+
+	for i := 0; i < nodeNumber; i++ {
+
+		zone, err := getZoneIDByName(cs, "ch-dk-2")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		template, err := getTemplateIDByName(cs, "Linux Ubuntu 18.04 LTS", zone)
+		if err != nil {
+			log.Fatal(err)
+		}
```

Comment:
the cloud-init says xenial, I'd keep Xenial and let the user choose the region. We had this issue with Rancher before and it's been a pain.